src/cljs/athens/router.cljs

```diff
@@ -233,3 +231,37 @@
     router
     on-navigate
     {:use-fragment true}))
+
+
+(rf/reg-event-fx
+  :init-routes!
+  (fn [_ _]
+    (init-routes!)
+    {}))
+
+
+;; Permalink param processing
+
+(def graph-param-key "graph")
+
+
+(defn consume-graph-param
+  "Removes and returns the graph-id in the current URL, if any."
+  []
+  ;; Note: don't use the reitit.frontend functions here, as the router
+  ;; it not yet initialized during boot.
+  (let [url       (js/URL. js/window.location)
+        graph-id  (.. url -searchParams (get graph-param-key))]
+    ;; Replace history with a version without the graph param.
+    (.. url -searchParams (delete graph-param-key))
+    (js/history.replaceState js/history.state nil url)
+    graph-id))
+
+
+(defn create-url-with-graph-param
+  "Create a URL containing graph-id."
+  [graph-id]
+  (let [url (js/URL. js/window.location)]
```

Comment:
this is weird in case of electron, because `file:///Users/neotyk/src/athens/resources/public/index.html?graph=https%3A%2F%2Fteam.athensresearch.org#/`

Reply:
I should remove the permalink button on electron, did so now.